chainercv/links/model/deeplab/deeplab_v3_plus.py

```diff
@@ -0,0 +1,278 @@
+from chainercv.links.model.deeplab.aspp import SeparableASPP
+from chainercv.links.model.deeplab.xception import Xception65
+from chainercv.transforms import resize
+from chainercv import utils
+
+import numpy as np
+
+
+class Decoder(chainer.Chain):
+
+    """Decoder for DeepLab V3+.
+
+    Args:
+        in_channels (int): Number of channels of input arrays.
+        out_channels (int): Number of channels of output arrays.
+        proj_channels (int): Number of channels of output of
+            first 1x1 convolution.
+        depth_channels (int): Number of channels of output of
+            convolution after concatenation.
+        bn_kwargs (dict): Keyword arguments passed to initialize
+            :class:`chainer.links.BatchNormalization`.
+
+    """
+
+    def __init__(self, in_channels, out_channels, proj_channels=48,
+                 depth_channels=256, bn_kwargs={}):
+        super(Decoder, self).__init__()
+
+        with self.init_scope():
+            self.feature_proj = Conv2DBNActiv(in_channels, proj_channels, 1)
+            concat_channels = in_channels+proj_channels
+            self.conv1 = SeparableConv2DBNActiv(
+                concat_channels, depth_channels, 3, 1, 1, 1, nobias=True,
+                dw_activ=F.relu, pw_activ=F.relu, bn_kwargs=bn_kwargs)
+            self.conv2 = SeparableConv2DBNActiv(
+                depth_channels, depth_channels, 3, 1, 1, 1, nobias=True,
+                dw_activ=F.relu, pw_activ=F.relu, bn_kwargs=bn_kwargs)
+            self.conv_logits = L.Convolution2D(
+                depth_channels, out_channels, 1, 1, 0)
+
+    def __call__(self, x, pool):
+        x = self.feature_proj(x)
+        pool = F.resize_images(pool, x.shape[2:])
+        h = F.concat((pool, x), axis=1)
+        h = self.conv1(h)
+        h = self.conv2(h)
+        logits = self.conv_logits(h)
+
+        return logits
+
+
+class DeepLabV3plus(chainer.Chain):
+
+    """Base class of DeepLab V3+.
+
+    Args:
+        fature_extractor (callable): feature extractor network.
+        aspp (callable): ASPP network.
+        decoder (callable): decoder network.
+        crop (tuple of ints): minimum image size of inputs.
+            if height or width is lower than this values,
+            input images are padded to be this shape.
+            The default value is :obj:`(513, 513)`
+
+    """
+
+    def __init__(self, feature_extractor, aspp, decoder,
+                 crop=(513, 513), scales=[1.0], flip=False):
+        super(DeepLabV3plus, self).__init__()
+
+        if not isinstance(crop, (list, tuple)):
+            crop = (int(crop), int(crop))
+        self.crop = crop
+        self.scales = scales
+        self.flip = flip
+
+        with self.init_scope():
+            self.feature_extractor = feature_extractor
+            self.aspp = aspp
+            self.decoder = decoder
+
+        self.feature_extractor.pick = 'entryflow_block2', 'exitflow_block2'
```

Comment:
Can you use `PickableSequantialChain` for the feature extractor?

Reply:
@yuyu2172 
What does this mean?
If you mean about implementing Xception65 by PickableSequentialChain, it may be not able.
Because DeepLab v3+ needs intermediate output of intermediate XceptionBlock of Xception65, I couldn't implement that.

Anyway, this line is redundant. already removed.

Reply:
Sorry. This was my mistake.
Deleting this line is ok.